src/library/commissioning_session.cpp

```diff
@@ -181,22 +181,22 @@ void CommissioningSession::HandleJoinFin(const coap::Request &aJoinFin)
     VerifyOrExit((vendorStackVersionTlv = tlvSet[tlv::Type::kVendorStackVersion]) != nullptr, error = Error::kNotFound);
     VerifyOrExit(vendorStackVersionTlv->IsValid(), error = Error::kBadFormat);
 
-    LOG_INFO("received JOIN_FIN.req: vendorName={}, vendorModel={}, vendorSWVversion={}, "
-             "vendorStackVersion={}",
-             vendorNameTlv->GetValueAsString(), vendorModelTlv->GetValueAsString(),
-             vendorSwVersionTlv->GetValueAsString(), utils::Hex(vendorStackVersionTlv->GetValue()));
-
     if (auto provisioningUrlTlv = tlvSet[tlv::Type::kProvisioningURL])
     {
         auto vendorDataTlv = tlvSet[tlv::Type::kVendorData];
         VerifyOrExit(vendorDataTlv != nullptr, error = Error::kNotFound);
         VerifyOrExit(vendorDataTlv->IsValid(), error = Error::kBadFormat);
-        VerifyOrExit(provisioningUrlTlv->GetValueAsString() == mJoinerInfo.mProvisioningUrl, error = Error::kReject);
```

Comment:
We need to at least add this to the `CommisioningHandler` in `CommissionerApp`:
https://github.com/openthread/ot-commissioner/blob/4f63a0f9bbfe9db2106ea8decf784a4399ef67a2/src/app/commissioner_app.cpp#L52

You may need to change the static default handle to a member function of class `CommissionerApp` now.

Reply:
Changed the CommissionerApp to only accept configured provisioning urls